pkg/controller/install/certresources.go

```diff
@@ -215,21 +215,21 @@ func (i *StrategyDeploymentInstaller) installCertRequirementsForDeployment(deplo
 	}
 	service.SetName(ServiceName(deploymentName))
 	service.SetNamespace(i.owner.GetNamespace())
-	ownerutil.AddNonBlockingOwner(service, i.owner)
```

Comment:
Should we just make AddNonBlockingOwner idempotent so it's safe to call multiple times with the same object/owner pair?

Reply:
IIRC it is idempotent, the problem here was the fact that we added, then did an append below